tempesta_fw/http_sess_conf.c

```diff
@@ -159,12 +159,33 @@ tfw_cfgop_cookie_set_options(TfwStickyCookie *sticky, const char *options)
 		T_ERR_NL("http_sess: too long cookie options length.\n");
 		return -EINVAL;
 	}
-	sticky->options.data = sticky->options_str;
-	sticky->options.len = len + 2;
 
-	sticky->options_str[0] = ';';
-	sticky->options_str[1] = ' ';
-	memcpy(&sticky->options_str[2], options, len);
+	sticky->options_str[sticky->options.len + 0] = ';';
+	sticky->options_str[sticky->options.len + 1] = ' ';
+	memcpy(&sticky->options_str[sticky->options.len + 2], options, len);
+
+	sticky->options.len += len + 2;
+
+	return 0;
+}
+
+int
+tfw_cfgop_cookie_set_path(TfwStickyCookie *sticky, const char *path_val)
+{
+	static const char *path = "path=";
+	size_t path_val_len = strlen(path_val);
+	if (path_val_len + 2 > STICKY_OPT_MAXLEN) {
```

Comment:
Before the patch `tfw_cfgop_cookie_set_options()` was the only place where `sticky->options_str` was modified. The options was a string with both value names and values, thus `strlen(options) + 2 > STICKY_OPT_MAXLEN` worked fine. (The same argument is not allowed to be listed multiple times, thanks to `cfg.c`).

But now the string is modified in three places: when parsing `option`, when parsing `path` and on default `path` value. Only first two have free space check for the buffer, and all of them don't account data already pushed to buffer.